espree.js

```diff
@@ -124,8 +131,8 @@ export function tokenize(code, options) {
 /**
  * Parses the given code.
  * @param {string} code The code to tokenize.
- * @param {Object} options Options defining how to tokenize.
- * @returns {ASTNode} The "Program" AST node.
+ * @param {ParserOptions} options Options defining how to tokenize.
+ * @returns {acorn.Node} The "Program" AST node.
```

Comment:
Objects returned from `parse()` also have `sourceType` property, and (optionally) `comments` and `tokens` properties.

https://github.com/eslint/espree/blob/b578a66991985d96d5e6ee4f388c4356ad0b3594/lib/espree.js#L154-L161